### PR TITLE
Tegra denver plat support v1

### PR DESCRIPTION
**Pull request by vwadekar**
Support for Tegra's "Project Denver" CPU platforms

### CODE REVIEW

**Comment by danh-arm** [comment]
Can one of the admins verify this patch?

File: plat/nvidia/tegra/common/tegra_sip_calls.c [hunk 1]
```diff
@@ -85,6 +95,24 @@ uint64_t tegra_sip_handler(uint32_t smc_fid,
 
 		SMC_RET1(handle, 0);
 
+	case TEGRA_SIP_AARCH_SWITCH:
```
Comment (athoelke) [inline plat/nvidia/tegra/common/tegra_sip_calls.c]: This is a SMC32 call - based on the smc_fid value. Thus you should clean the high bits of the x1/x2 parameters
`x1 = (uint32_t)x1;`
etc. before using them. In the case that a call came from AArch32, it is not guaranteed that the high bits are zero, which this code appears to depend on.

I notice that this also affects the VIDEOMEM_REGION function, but I can't add a comment there in this commit diff.

 I presume you already realise that these two SIP service calls cannot be passed addresses above 4GB as they are 32-bit functions.

File: plat/nvidia/tegra/common/tegra_sip_calls.c [hunk 2]
```diff
@@ -38,7 +38,14 @@
 #include <runtime_svc.h>
 #include <tegra_private.h>
 
+#define NS_SWITCH_AARCH32	1
+#define SCR_RW_BITPOS		10
```
Comment (athoelke) [inline plat/nvidia/tegra/common/tegra_sip_calls.c]: You could define this in terms of the SCR_RW_BIT definition by using a GCC builtin function:
`#define SCR_RW_BITPOS __builtin_ctz(SCR_RW_BIT)`
This is the technique used in the GIC drivers to implement the `interrupt_type_to_line()` functionality. (I'm actually wondering if we should have a less obscure macro to do this?

**Comment by vwadekar** [comment]
New PR - #340

**Pull request closed without merging**